tests/TestCase/View/Helper/PaginatorHelperTest.php

```diff
@@ -3495,4 +3495,35 @@ protected function setPagingParams($params, bool $merge = true)
         }
         $this->View->setRequest($this->View->getRequest()->withAttribute('paging', $params));
     }
+
+    /**
+     * Test that generated URLs work without sort defined within the request
+     *
+     * @return void
+     */
+    public function testPaginationWithPassedParams()
+    {
+        $request = new ServerRequest([
+            'url' => '/articles/index/whatever/3',
+            'params' => [
+                'plugin' => null,
+                'controller' => 'articles',
+                'action' => 'index',
+                'pass' => ['whatever', '3'],
+            ],
+            'base' => '',
+            'webroot' => '/',
+        ]);
+        Router::setRequest($request);
+
+        $this->View->setRequest($request->withAttribute('paging', [
+            'Article' => [
+                'scope' => 'article',
+            ],
+        ]));
+        $this->Paginator = new PaginatorHelper($this->View);
+        $result = $this->Paginator->generateUrl(['page' => 2]);
+        $expected = '/articles/index/whatever/3?article%5Bpage%5D=2';
+        static::assertSame($expected, $result);
```

Comment:
```suggestion
        $this->assertSame($expected, $result);
```
is now this should be used

Reply:
@dereuromark I changed it. There is no problem in this context, but it is not good practice to call a static method dynamically [https://github.com/sebastianbergmann/phpunit/blob/9.5/src/Framework/Assert.php#L1327](https://github.com/sebastianbergmann/phpunit/blob/9.5/src/Framework/Assert.php#L1327)

Reply:
Well, it is PHPUnit's official documentation and standpoint to use them non-statically :)

Reply:
[https://phpunit.readthedocs.io/en/9.5/assertions.html#appendixes-assertions](https://phpunit.readthedocs.io/en/9.5/assertions.html#appendixes-assertions)
the both uses are right :)

Reply:
Not if you follow the rest of their docs :)

Reply:
Totally agree with you